### ISSUE TITLE
InputStreamFileSystem (and possibly others) do not set filename when throwing FileNotFoundException

### ISSUE THREAD

**Issue by Pwootage** [MightyPirates/OpenComputers#1563]
I noticed this while implementing the filesystem for my OS (made it very tricky to track down which file it wasn't finding).

https://github.com/MightyPirates/OpenComputers/blob/master-MC1.7.10/src/main/scala/li/cil/oc/server/fs/InputStreamFileSystem.scala#L39-L41

``` scala
  override def open(path: String, mode: Mode) = this.synchronized(if (mode == Mode.Read && exists(path) && !isDirectory(path)) {
    val handle = Iterator.continually((Math.random() * Int.MaxValue).toInt + 1).filterNot(handles.contains).next()
    openInputChannel(path) match {
      case Some(channel) =>
        handles += handle -> new Handle(this, handle, path, channel)
        handle
      case _ => throw new FileNotFoundException()
    }
  } else throw new FileNotFoundException())
```

I can look for the rest of these and open a pull request, if you like (after finals tomorrow :P)

**Comment by fnuecke** [comment]
Ah, you mean throwing something saying "too many open handles" instead? Sure, if there are other instances (can't remember just how re-used the code in the FS stuff is), gladly!

See if just adding a message to the exception does the trick so stuff depending on it being a FNFE don't break, if it doesn't oh well.

**Comment by Pwootage** [comment]
Fixed by #1570
